src/components/molecules/UserProfilePicture/UserProfilePicture.tsx

```diff
@@ -16,154 +17,176 @@ import {
+  }
+
+  return DEFAULT_PROFILE_IMAGE;
+};
+
 export interface UserProfilePictureProp {
   user: WithId<User>;
   isAudioEffectDisabled?: boolean;
   miniAvatars?: boolean;
   avatarClassName?: string;
   avatarStyle?: object;
   containerStyle?: object;
-  reactionPosition?: "right" | "left" | undefined;
+  reactionPosition?: "left" | "right";
 }
 
 // @debt This component should be divided into a few with simpler logic. Also, remove `styled components`
 // @debt the UserAvatar component serves a very similar purpose to this, we should unify them as much as possible
-const UserProfilePicture: React.FC<UserProfilePictureProp> = ({
+export const UserProfilePicture: React.FC<UserProfilePictureProp> = ({
   isAudioEffectDisabled = true,
   miniAvatars = false,
-  avatarClassName = "profile-icon",
-  avatarStyle,
-  containerStyle,
-  reactionPosition,
+  avatarClassName = "UserProfilePicture__avatar",
+  avatarStyle, // TODO: do we need this prop? Can we remove it? Or at least rename it? Only seems to be used in MapPartygoersOverlay
+  containerStyle, // TODO: do we need this prop? Can we remove it? Or at least rename it? Only seems to be used in MapPartygoersOverlay
+  reactionPosition = "right",
   user,
 }) => {
-  // @debt some of the redux patterns exist for this, but I don't believe anything actually uses them/calls this at the moment
-  const muteReactions = useSelector((state) => state.room.mute);
-  const { openUserProfileModal } = useProfileModalControls();
-
-  const [pictureUrl, setPictureUrl] = useState("");
-
-  const randomAvatarUrl = (id: string) =>
-    "/avatars/" +
-    RANDOM_AVATARS[Math.floor(id?.charCodeAt(0) % RANDOM_AVATARS.length)];
-
-  const avatarUrl = useCallback(
-    (id: string, anonMode?: boolean, pictureUrl?: string) => {
-      if (anonMode || !id) {
-        return setPictureUrl(DEFAULT_PROFILE_IMAGE);
-      }
-
-      if (!miniAvatars && pictureUrl) {
-        return setPictureUrl(pictureUrl);
-      }
-
-      if (miniAvatars) {
-        return setPictureUrl(randomAvatarUrl(id));
-      }
-
-      return setPictureUrl(DEFAULT_PROFILE_IMAGE);
-    },
-    [miniAvatars]
-  );
+  const venueId = useVenueId();
 
-  useEffect(() => {
-    avatarUrl(user.id, user.anonMode, user.pictureUrl);
-  }, [avatarUrl, user.anonMode, user.id, user.pictureUrl]);
+  const { openUserProfileModal } = useProfileModalControls();
 
-  const venueId = useVenueId();
+  // TODO: extract this logic into a generic custom reactions component + improve it?
   const reactions = useReactions(venueId);
-
+  // @debt some of the redux patterns exist for this, but I don't believe anything actually uses them/calls this at the moment. Used in MapPartygoersOverlay
+  const muteReactions = useSelector((state) => state.room.mute);
   const typedReaction = reactions ?? [];
-
   const messagesToBand = typedReaction.find(
     (r) => r.reaction === "messageToTheBand" && r.created_by === user.id
   ) as MessageToTheBandReaction | undefined;
```

Comment:
[TODO] extract this logic into a generic custom reactions component + improve it?

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1323/commits/2dc683db970609e234f3abb99b1a70d4485af6e8